src/main/java/vmware/samples/vcenter/helpers/FolderHelper.java

```diff
@@ -38,24 +38,32 @@
     public static String getFolder(
         StubFactory stubFactory, StubConfiguration sessionStubConfig,
         String datacenterName, String folderName) {
-
-        // Get the datacenter
-        Set<String> datacenters = Collections.singleton(DatacenterHelper
-            .getDatacenter(stubFactory, sessionStubConfig, datacenterName));
-
-        // Get the folder
+    	
+    	// Get the folder
         Folder folderService = stubFactory.createStub(Folder.class,
             sessionStubConfig);
         Set<String> vmFolders = Collections.singleton(folderName);
-        FolderTypes.FilterSpec vmFolderFilterSpec =
-                new FolderTypes.FilterSpec.Builder().setNames(vmFolders)
-                    .setDatacenters(datacenters)
-                    .build();
+        FolderTypes.FilterSpec.Builder vmFolderFilterSpecBuilder =
+                new FolderTypes.FilterSpec.Builder().setNames(vmFolders);
+                    
+        if(null != datacenterName)
+        {	
+        	// Get the datacenter
+        	Set<String> datacenters = Collections.singleton(DatacenterHelper
+            .getDatacenter(stubFactory, sessionStubConfig, datacenterName));
+        	vmFolderFilterSpecBuilder.setDatacenters(datacenters);
+        }
+        
         List<FolderTypes.Summary> folderSummaries = folderService.list(
-            vmFolderFilterSpec);
+        		vmFolderFilterSpecBuilder.build());
         assert folderSummaries.size() > 0 : "Folder " + folderName
-                                            + "not found in datacenter: "
+                                            + " not found in datacenter: "
                                             + datacenterName;
         return folderSummaries.get(0).getFolder();
     }
+
+	public static String getFolder(StubFactory stubFactory, StubConfiguration sessionStubConfig, String vmFolderName) {
```

Comment:
We have followed a 80 char line length limit in all the samples. Lets keep it consistent to any new changes as well.